File: client-app/src/app/models/product.ts
export interface IProduct{
    id: string ;
    name: string;
    brand: string;
    description: string;
    category: string;
    price: string;
    photo: string;
}

// export interface IProductFormValues extends Partial<IProduct> {
//     price?: string;
// }

// export class ProductFormValues implements IProductFormValues {
//     id?: string = undefined;
//     name: string = '';
//     brand: string = '';
//     description: string = '';
//     category: string = '';
//     price: string = '';
//     photo: string = '';

//     constructor(init?: IProductFormValues) {
//         if(init){
            
//         }
//         Object.assign(this, init);
        
        
//     }
// }

File: Domain/Product.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Product
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Brand { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public double Price { get; set; }
        public string Photo { get; set; }

        public ICollection<Distribution> Distributions {get; set;}
        public ICollection<StockRequest> StockRequests {get; set;}
    }
}